aicsimageio/readers/reader.py

```diff
@@ -149,7 +151,24 @@ def set_scene(self, id: int):
         ------
         IndexError: the provided scene id is not found in the available scene id list
         """
-        pass
+        # Only need to run when the scene id is different from current scene
+        if id != self.current_scene:
```

Comment:
Might be best to change this from `id` to `id_` since `id` is a builtin function

Reply:
I _always_ forget this.